### ISSUE TITLE
provide batch set get delete method

### ISSUE THREAD

**Issue by Wang-Jun-Chao** [eko/gocache#61]
In many case, we need handle data in batch. Can you provide batch set,get and delete method .

**Comment by sgtsquiggs** [comment]
this would be useful for the redis cache since the bulk redis commands are a _lot_ faster than individual

**Comment by paulo-raca** [comment]
+1.
We also need to be able to provide bulk-load function for LoadableCaches

**Comment by yagnasrinath** [comment]
+1

**Comment by eko** [comment]
Agreed that this is worth having — bulk commands are genuinely much faster on Redis and Memcache, and a bulk load function for `LoadableCache` is a real need too.

I have not implemented it yet because the API needs to be right first, and there are a few things to settle:

- what `GetMulti` returns for the keys that are missing — a partial map, or a per-key error,
- whether `SetMulti` is atomic (it cannot be on most stores) and how partial failures are reported,
- what stores without a bulk primitive do: loop, or report unsupported the way `SetIfNotExists` does now with `store.ErrNotSupported`,
- how it composes with `Chain`, which has to know which keys were found in which layer to put them back.

Leaving this open. If someone wants to take it on, a proposal on those four points would be the useful first step — I would rather agree on the shape before code, since this ends up in `CacheInterface`.
